rugby.ipynb

```diff
@@ -0,0 +1,3579 @@
+{
```

Comment:
<p>This cell should be:</p><p> </p><pre class="ql-syntax" spellcheck="false">with model:
&nbsp;&nbsp;&nbsp; prior = pm.sample_prior_predictive()
&nbsp;&nbsp;&nbsp; trace = pm.sample(500, tune=500, cores=4)
&nbsp;&nbsp;&nbsp; posterior_predictive = pm.sample_posterior_predictive(trace)
</pre><p> </p><p>I'll try to explain the reason behind each change. </p><p> </p><p>Continuing on the comment above, to sample from the prior and thus get the data that goes into prior and prior predictive groups, <code>sample_prior_predictive</code>  has to be used, sadly the current code does not generate the variables for the prior and prior predictive groups.</p><p> </p><p>The current rugby example has 4 chains and 500 draws whereas the current code generates 1000 draws and 3 chains. The changes above should fix that difference</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/arviz-devs/arviz_example_data/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Reply:
<p>I made the above changes in the cell however while executing the code, I am facing a new error <code>ValueError: lam value too large</code> for the code line <code>prior = pm.sample_prior_predictive()</code> .</p><p> </p><p>On searching for the error on google, I found that the error is due to certain numpy libraries used in pm.sample_prior_predictive() method. Moreover, this type of error is faced on Windows operating system. </p>

Reply:
Can you try:

```
home = pm.Normal('home', mu=0, sigma=1)
intercept = pm.Normal('intercept', mu=3, sigma=1)
```

This should be more restrictive and fix the overflow issue you mentioned above

Reply:
<p>Unfortunately, even these changes are not working out. Program is encountering the same error. Is there any other workaround? </p>

Reply:
Maybe with

```
sd_att = pm.HalfStudentT('sd_att', nu=3, sigma=1.5)
sd_def = pm.HalfStudentT('sd_def', nu=3, sigma=1.5)
```

or if this does not work either, with 

```
sd_att = pm.HalfNormal('sd_att', sigma=2)
sd_def = pm.HalfNormal('sd_def', sigma=2)
```

After this, push the changes independently of the code working or not. It it were still not working I'd play around to see what can be happening.

Reply:
I tried with the above code lines and the code 

`sd_att = pm.HalfNormal('sd_att', sigma=2)
sd_def = pm.HalfNormal('sd_def', sigma=2)` 

worked perfectly fine. The code prior to these lines still had an error. However, now I am facing another issue with the `arviz.from_pymc3(...., dims = {...})`. The code `dims={"atts": [teams], "defs": [teams], "atts_star": [teams], "defs_star": [teams]}` is throwing an error `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`. However, the teams are not boolean arrays but still the code is treating it as a boolean array.

Reply:
Oh, my bad, `dims` is a mapping from variable names to dimension names, whereas now it is kind of a mapping between variable names and coordinate values. The correct syntax should be:

    dims={"atts": ["team"], "defs": ["team"], "atts_star": ["team"], "defs_star": ["team"]}

Note that the dimension name is `team` which is the key in `coords` (coords sets the values of `team` dimension to the values in array `teams`). Also, the variables whose dimension is `match` (home and away points, check the observed data group of the current example) should also be added to `dims`. I hope the pattern is clear now, sorry about the previous mix-up

Reply:
I modified the dims part as below:

`dims = {"atts": ["team"], "defs": ["team"], "atts_star": ["team"],  "defs_star": ["team"], "home_points": ["match"], "away_points": ["match"]}`

However, the ["team"] value is posing an error `ValueError: dimensions ('team',) must have the same length as the number of data dimensions, ndim=2`. Am I missing something in the modified `dims` definition?

Reply:
Can you push the current code?